kernel/main.c

```diff
@@ -41,11 +42,18 @@
 #include <3rd_party/acpi/acpi.h>
 #include <3rd_party/acpi/acexcep.h>
 
+#define EFI_DEBUG 0
+#include <efi.h>
+#include <efiapi.h>
+
 static bool read_multiboot_info_passed;
 static u32 saved_multiboot_magic;
 static multiboot_info_t *saved_multiboot_mbi;
 static void read_multiboot_info(void);
 
+static bool booting_with_uefi;
```

Comment:
After some thinking, I really don't like putting UEFI stuff here like that. It's much better to create a kernel/uefi.c file and put everything there. Also, the `booting_with_uefi` is not necessary. You can simply check for `if (!uefi_rt_addr)` in the code.

Reply:
Also, introduce the `include/tilck/kernel/uefi.h` header file for putting all the declarations for the uefi-related functions. For the coding style, check the other headers.